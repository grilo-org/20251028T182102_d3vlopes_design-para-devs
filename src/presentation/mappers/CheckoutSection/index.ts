import { CheckoutSectionAPI } from '@/api/models/Sections';
import { CheckoutSectionModel } from '@/domain/models/Sections';

export class CheckoutSectionMapper {
	static toDomain(api: CheckoutSectionAPI): CheckoutSectionModel {
		return {
			id: api.sectionID,
			heading: api.title,
			image: {
				src: api.image.url,
				alt: String(api.image.imageDescription),
			},
			originalPrice: api.originalPrice,
			salePrice: api.salePrice,
			button: {
				text: api.button.text,
				href: api.button.target,
			},
			icons: api.icons.map((icon) => {
				return {
					id: icon.id,
					name: icon.name,
					src: icon.image.url,
					alt: String(icon.image.imageDescription),
				};
			}),
		};
	}
}
