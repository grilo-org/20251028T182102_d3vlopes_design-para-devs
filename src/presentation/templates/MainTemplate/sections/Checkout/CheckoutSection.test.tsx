import { render, screen } from '@testing-library/react';

import { checkoutSectionMock as mock } from './CheckoutSection.mock';

import { CheckoutSection } from '.';

const { getByRole, getByText, getByTestId } = screen;

jest.mock('@/presentation/components/molecules/PaymentIcons', () => ({
	PaymentIcons: () => <div data-testid="mock-payment-icons" />,
}));

describe('<CheckoutSection />', () => {
	beforeEach(() => render(<CheckoutSection {...mock} />));

	it('should contain correct section id', () => {
		const { container } = render(<CheckoutSection {...mock} />);

		const section = container.getElementsByTagName('section')[0];

		expect(section).toHaveAttribute('id', mock.id);
	});

	it('should render image', () => {
		const image = getByRole('img', {
			name: mock.image.alt,
		});

		expect(image).toBeInTheDocument();
	});

	it('should render section heading', () => {
		const heading = getByRole('heading', {
			name: mock.heading,
		});

		expect(heading).toBeInTheDocument();
	});

	it('should render original price', () => {
		const originalPrice = getByText(/R\$ 60,00/i);

		expect(originalPrice).toBeInTheDocument();
	});

	it('should render sale price', () => {
		const salePrice = getByText(/R\$ 9,90/i);

		expect(salePrice).toBeInTheDocument();
	});

	it('should render button', () => {
		const button = getByRole('button', { name: mock.button.text });

		expect(button).toBeInTheDocument();
	});

	it('should render anchor with correct href value', () => {
		const anchor = getByRole('link');

		expect(anchor).toHaveAttribute('href', mock.button.href);
	});

	it('should render PaymentIcons', () => {
		const paymentIcons = getByTestId('mock-payment-icons');
		const text = getByText('Formas de pagamento');

		expect(paymentIcons).toBeInTheDocument();
		expect(text).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<CheckoutSection {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
