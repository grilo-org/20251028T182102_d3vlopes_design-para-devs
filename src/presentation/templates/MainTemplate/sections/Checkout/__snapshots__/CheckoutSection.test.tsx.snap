// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutSection /> should match snapshot 1`] = `
<div>
  <section
    class="bg-background-400 lg:rounded-[8px]"
    id="checkout"
  >
    <div
      class="flex flex-col items-center py-[3.25rem] lg:hidden"
    >
      <h2
        class="font-heading font-bold text-[2.5rem] leading-[2.5rem] text-complementary-white text-center w-[23.4375rem] lg:text-left lg:text-6xl lg:leading-[2.25rem] xl:text-9xl xl:leading-[2.875rem] xl:w-[34.6875rem]"
      >
        Se diferencie como dev, comece a estudar design agora!
      </h2>
      <div
        class="relative w-[15.3125rem] h-[21.3125rem] mt-5 mb-11 lg:mt-0l g:mb-0 lg:w-[23.125rem] lg:h-[31.125rem]"
      >
        <img
          alt="Capa do eBook com um desenvolvedor em frente ao computador, escrevendo cÃ³digo"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FNRDgi138ReqCWdYg7l1j&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="flex flex-col justify-start md:my-4"
      >
        <span
          class="font-highlight text-2xl leading-[1.5rem] text-neutral-50 font-normal"
        >
          De
           
          <strong
            class="line-through text-complementary-red font-normal"
          >
            R$ 
            60,00
          </strong>
           
          por apenas
        </span>
        <strong
          class="relative left-[-2px] font-highlight font-bold text-12xl leading-[4.5rem] text-complementary-green max-w-max"
        >
          R$ 
          9,90
        </strong>
      </div>
      <a
        class="mt-3 mb-6"
        href="https://pay.hotmart.com/Y88426530P?checkoutMode=10&offDiscount=LANCAMENTO"
        rel="noreferrer"
        target="_blank"
      >
        <button
          class="font-heading inline-flex items-center justify-center whitespace-nowrap ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-gradient-primary rounded-[8px] py-6 font-bold tracking-wider w-[17.1875rem] text-xxl lg:w-[28rem] lg:text-5xl xl:w-[34.6875rem]"
        >
          Quero aprender
        </button>
      </a>
      <div
        class="flex flex-col gap-3 items-center lg:mt-6"
      >
        <p
          class="font-highlight font-bold text-neutral-100 lg:text-lg"
        >
          Formas de pagamento
        </p>
        <div
          data-testid="mock-payment-icons"
        />
      </div>
    </div>
  </section>
</div>
`;
